source/portable/NetworkInterface/libslirp/MBuffNetworkInterface.c

```diff
@@ -77,76 +77,92 @@ extern void vMBuffNetifBackendDeInit( void * pvBackendContext );
 
 static void vNetifReceiveTask( void * pvParameters );
 
-MBuffNetDriverContext_t xDriverCtx = { 0 };
-
 /**
  * @brief Initialize the MessageBuffer backed network interface.
  *
  * @return BaseType_t pdTRUE on success
  */
-BaseType_t xNetworkInterfaceInitialise( void )
+BaseType_t xNetworkInterfaceInitialise( NetworkInterface_t * pxNetif )
 {
     BaseType_t xResult = pdTRUE;
 
-    if( xDriverCtx.xInterfaceState == pdFALSE )
+    pxNetif->pvArgument = pvPortMalloc( sizeof( MBuffNetDriverContext_t ) );
+
+    if( pxNetif->pvArgument == NULL )
     {
-        xDriverCtx.xInterfaceState = pdFALSE;
+        FreeRTOS_printf( ( "Failed to allocate memory for pxNetif->pvArgument" ) );
```

Comment:
Would prefer that this function return pdFALSE on failure, rather than asserting.

Reply:
If it returns pdFALSE will the upstream application be graceful about it?

configASSERT( pxNetif->pvArgument != NULL ); might also be OK

Crashing here with an error is OK.

Reply:
I would like that too. Not blocking this PR for that though.

Because a `configASSERT( 0 )` will also trigger compiler warnings as `configASSERT` compares the variable with `0` and comparing `0` with `0` will lead to a "always taking the same path : dead code" warning.